src/test/java/guitests/PanelRenameTest.java

```diff
@@ -0,0 +1,120 @@
+package guitests;
+
+import javafx.scene.input.KeyCode;
+
+import org.junit.Test;
+import org.loadui.testfx.utils.FXTestUtils;
+
+import prefs.Preferences;
+import ui.UI;
+import ui.components.PanelNameTextField;
+import util.PlatformEx;
+import util.events.ShowRenamePanelEvent;
+
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+public class PanelRenameTest extends UITest {
+
+    public static final int EVENT_DELAY = 500;
+
+    @Override
+    public void launchApp() {
+        FXTestUtils.launchApp(TestUI.class, "--testconfig=true", "--bypasslogin=true");
+    }
+
+    @Test
+    public void panelRenameTest() {
+        
+        // Test for saving panel information with corner cases
+        
+        // maximize
+        press(KeyCode.CONTROL).press(KeyCode.X).release(KeyCode.X).release(KeyCode.CONTROL);
+        sleep(EVENT_DELAY);
+
+        // Testing rename cancel
```

Comment:
Comments should be "well written". Comments like this are a bit too cryptic for an outsider. Try to improve. e.g. `// Rename, but press ESC midway to cancel`